src/com/facebook/buck/jvm/java/DefaultJavaLibrary.java

```diff
@@ -319,6 +321,12 @@ private static Path getOutputJarDirPath(BuildTarget target) {
     return BuildTargets.getGenPath(target, "lib__%s__output");
   }
 
+  private Optional<SourcePath> sourcePathForOutputJar() {
+    return outputJar.isPresent()
+        ? Optional.<SourcePath>of(new BuildTargetSourcePath(getBuildTarget(), outputJar.get()))
+        : Optional.<SourcePath>absent();
```

Comment:
It would be better to use `Optional.transform` here which handles the `absent` case already.